modules/common/math/mpc_solver.cc

```diff
@@ -67,10 +67,17 @@ bool SolveLinearMPC(const Matrix &matrix_a, const Matrix &matrix_b,
 
   Matrix matrix_k =
       Matrix::Zero(matrix_b.rows() * horizon, matrix_b.cols() * horizon);
-  for (unsigned int r = 0; r < horizon; ++r) {
+  matrix_k.block(0, 0, matrix_b.rows(), matrix_b.cols()) = matrix_b;
+  for (unsigned int r = 1; r < horizon; ++r) {
     for (unsigned int c = 0; c <= r; ++c) {
```

Comment:
recommand `for (unsigned int c = 0; c < r; ++c)`, then write c == r case outside of for loop.